authz.go

```diff
@@ -40,15 +40,14 @@ func PermissionAllowed(p Permission, ps []Permission) bool {
 	}
 
 	for _, perm := range ps {
-		permID := ID(0)
-		if perm.ID != nil {
-			permID = *perm.ID
-			if !permID.Valid() {
-				return false
+		if perm.ID == nil {
+			if perm.Action == p.Action && perm.Resource == p.Resource && perm.OrgID == p.OrgID {
+				return true
+			}
+		} else {
+			if (*perm.ID).Valid() && *perm.ID == pID && perm.Action == p.Action && perm.Resource == p.Resource && perm.OrgID == p.OrgID {
```

Comment:
`(*perm.ID).Valid() && *perm.ID == pID`

Won't this always be false? pID is ID(0), which by definition is not valid:
```
// Valid checks whether the receiving ID is a valid one or not.
func (i ID) Valid() bool {
	return i != 0
}
```